presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -278,9 +278,10 @@ private void rebuildHashAggregationBuilder()
         if (hashAggregationBuilder != null) {
             hashCollisions += hashAggregationBuilder.getHashCollisions();
             expectedHashCollisions += hashAggregationBuilder.getExpectedHashCollisions();
+            hashAggregationBuilder.close();
         }
 
-        this.hashAggregationBuilder = new InMemoryHashAggregationBuilder(
+        hashAggregationBuilder = new InMemoryHashAggregationBuilder(
```

Comment:
seems so -- method first operates on `hashAggregationBuilder` (and this is what gets closed) and then assigns to `this.hashAggregationBuilder` -- same thing but looks different